Update doc about node removal in configuration v2 (#49748)
ydb-platform/ydb: ydb/docs/en/core/devops/configuration-management/configuration-v2/node-removal.md

Before starting the procedure, use the [Embedded UI](../../../reference/embedded-ui/ydb-monitoring.md#node_storage_page) to check that the affected storage groups are healthy, that is, all VDisks of these groups are shown in the `Ok` state (highlighted in green), with none in `Error` or `Degraded` state. The cluster must also have enough free slots to move the VDisks from the node while preserving the fault model: the remaining nodes must have free PDisk space and slots to accommodate the relocated VDisks without exceeding the number of failed fail domains/fail realms the configured erasure coding scheme can tolerate. For details on calculating the required capacity margin, see [{#T}](../../concepts/capacity-planning.md#hardware-estimation).

[SelfHeal](../../../maintenance/manual/selfheal.md) is enabled for dynamic groups by default. Before removing the node, make sure it is also enabled for any resources hosted on this node:

* If the node contains a static group VDisk, [enable static group SelfHeal](static-group-self-heal.md#on-off). Alternatively, you can move the static group VDisk off the node manually, see [{#T}](static-group-move.md).

> * If the node contains State Storage, Board, or SchemeBoard replicas, enable [Self Heal State Storage](../../../maintenance/manual/selfheal_statestorage.md#on-off). Alternatively, you can move these replicas off the node manually, see [{#T}](state-storage-move.md).

Reviewer: Ручная альтернатива ведёт на `state-storage-move.md`, вся английская версия которого скрыта под «Article under development» и не содержит законченной процедуры, тогда как RU ссылается на готовый `state-storage-reconfiguration.md`. Нужен один из двух вариантов: либо добавить английский перевод `state-storage-reconfiguration.md` и заменить ссылку на него, либо до появления перевода убрать ручную альтернативу из EN, оставив строку такой:

```markdown
* If the node contains State Storage, Board, or SchemeBoard replicas, enable [Self Heal State Storage](../../../maintenance/manual/selfheal_statestorage.md#on-off).
```

Reviewer 2: Да, почемуто не найду английскую версию state-storage-reconfiguration.md Надо ее добавить и сослаться на нее

To remove a static node:

1. If tablets are running on the node, [drain them](../../../maintenance/manual/node_restarting.md#replace-hardware).
1. [Check that the process can be stopped safely](../../../maintenance/manual/node_restarting.md#restart_process), then stop it.
1. Wait for SelfHeal to move the VDisks from the node. With the default settings, relocation starts approximately one hour after the node is stopped. To start the relocation immediately instead of waiting, mark the node's PDisks as `BROKEN` using [{{ ydb-short-name }} DSTool](../../../reference/ydb-dstool/index.md):

    ```bash

>     ydb-dstool -e <bs_endpoint> pdisk set --status BROKEN --pdisk-ids "[NodeId:PDiskId]"

Reviewer: После предыдущего шага узел уже остановлен, поэтому его PDisks недоступны через Whiteboard; согласно README `ydb-dstool`, для такого случая нужен `--unavail-as-offline`, иначе команда может отказаться помечать диски как `BROKEN`.

```suggestion
    ydb-dstool -e <bs_endpoint> pdisk set --status BROKEN --unavail-as-offline --pdisk-ids "[NodeId:PDiskId]"
```

To remove a dynamic node without affecting query processing:

1. [Drain the tablets](../../../maintenance/manual/node_restarting.md#replace-hardware) from the node and wait for the operation to complete.
1. Stop the dynamic node process.

After stopping the process, check the **Nodes** tab on the [cluster monitoring page](../../../reference/embedded-ui/ydb-monitoring.md#node_list_page) and verify that the node is no longer shown as connected.

## Removing a static node {#remove-static-node}

Static nodes serve the storage system and are listed in the `hosts` section. A static node can contain VDisks of dynamic and static groups, as well as State Storage, Board, and SchemeBoard replicas. These resources must be moved before the node is removed from the configuration.

Before starting the procedure, use the [Embedded UI](../../../reference/embedded-ui/ydb-monitoring.md#node_storage_page) to check that the affected storage groups are healthy. The cluster must also have enough free slots to move the VDisks from the node while preserving the fault model. For details on calculating the required capacity margin, see [{#T}](../../concepts/capacity-planning.md#hardware-estimation).

> [SelfHeal](../../../maintenance/manual/selfheal.md) is enabled for dynamic groups by default. If the node contains a static group VDisk, [enable static group SelfHeal](static-group-self-heal.md#on-off). If the node contains State Storage, Board, or SchemeBoard replicas, enable [Self Heal State Storage](../../../maintenance/manual/selfheal_statestorage.md#on-off).

Reviewer: Same issue as RU: the SelfHeal prerequisites need clearer structure. Currently "is enabled for dynamic groups by default. If the node contains..." confusingly mixes enabled/disabled state.

Should clearly show:
- Dynamic group SelfHeal: already enabled by default ✓
- Static group SelfHeal: **must be enabled if static group VDisks present**
- State Storage SelfHeal: **must be enabled if State Storage replicas present**

To remove a static node:

> 

Reviewer: Add WARNING block before the procedure steps. Proactively explain the operational time cost:

```suggestion
To remove a static node:

{% note warning %}

Removing a static node is a complex operation requiring relocation of many VDisks. **Expect approximately one hour of waiting with default SelfHeal settings** (VDisk relocation happens in the background). Actual time depends on:

- Amount of data on the node's VDisks
- Disk performance
- Current cluster load

You can **speed up the process** by monitoring relocation progress and manually moving remaining VDisks (see error handling section below).

Ensure the cluster has sufficient free slots on other nodes to accept the relocating VDisks.

{% endnote %}
```

This proactively addresses that node removal is operationally expensive without explanation.

1. If tablets are running on the node, [drain them](../../../maintenance/manual/node_restarting.md#replace-hardware).
1. [Check that the process can be stopped safely](../../../maintenance/manual/node_restarting.md#restart_process), then stop it.
1. Wait for SelfHeal to move the VDisks from the node. With the default settings, relocation starts approximately one hour after the node is stopped.
1. In the [Embedded UI](../../../reference/embedded-ui/ydb-monitoring.md#node_storage_page), check that no VDisks remain on the node and that the affected storage groups are healthy. If State Storage, Board, or SchemeBoard replicas were moved from the node, [check that the relocation is complete](../../../maintenance/manual/selfheal_statestorage.md#verify-result).
1. Fetch the current cluster configuration using the [ydb admin cluster config fetch](../../../reference/ydb-cli/commands/configuration/cluster/fetch.md) command:

    ```bash
    ydb [global options...] admin cluster config fetch > config.yaml
    ```

> 1. If the node being removed is not the last entry in the [`hosts`](../../../reference/configuration/hosts.md) list, preserve the IDs of the nodes that follow it: add `node_id` to each entry where it is not specified. Use the entry's current one-based position in the list as the value. If the last entry is being removed, this step is not required.

Reviewer: Needs concrete example to explain why preserving node_id matters:

```suggestion
1. If the node being removed is not the last entry in the [`hosts`](../../../reference/configuration/hosts.md) list, all following nodes' position indices will shift when you remove the entry. To preserve their original `node_id` values (which other configuration may reference), **explicitly set `node_id` for each following entry**. Use each node's current one-based position in the list. Example:

   Before removal (node 3 is being removed):
   ```yaml
   hosts:
     - host: node1
     - host: node2
     - host: node3    # ← removing this
     - host: node4
     - host: node5
   ```

   After removal (node_id 4 and 5 explicitly preserved):
   ```yaml
   hosts:
     - host: node1
     - host: node2
     - host: node4
       node_id: 4     # ← explicitly set (would shift to 3 without this)
     - host: node5
       node_id: 5     # ← explicitly set (would shift to 4 without this)
   ```

   If the last entry in the list is being removed, this step is not required.
```
